kola/tests/ignition/mount.go

```diff
@@ -0,0 +1,331 @@
+// Copyright 2019 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ignition
+
+import (
+	"strings"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+func init() {
+	// verify mount to `/var/lib/containers`
+	register.Register(&register.Test{
+		Name:        "coreos.ignition.mount.var-lib-containers",
+		Run:         mountVarLibContainers,
+		ClusterSize: 1,
+		UserData: conf.Ignition(`{
+			"ignition": {"version": "2.2.0"},
+			"storage": {
+				"disks": [
+					{
+						"device": "/dev/vda",
+						"wipeTable": false,
+						"partitions": [
+						{
+							"label": "CONTR",
+							"size": 0,
+							"start": 0
+						}
+						]
+					}
+				],
+				"filesystems": [
+					{
+						"mount": {
+							"device": "/dev/vda5",
+							"format": "xfs",
+							"label": "CONTR",
+							"wipeFilesystem": true
+						}
+					}
+				]
+			},
+			"systemd": {
+				"units": [
+					{
+						"name": "var-lib-containers.mount",
+						"enabled": true,
+						"contents": "[Mount]\nWhat=/dev/vda5\nWhere=/var/lib/containers\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+					}]
+			}
```

Comment:
Something else that'd be nice to test here is writing a file in each of those mounts and making sure that they were written when we boot.

Reply:
Sounds good, will update :+1: